airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/DefaultJdbcDatabase.kt

```diff
@@ -115,20 +115,29 @@ constructor(
         statementCreator: CheckedFunction<Connection, PreparedStatement, SQLException>,
         recordTransform: CheckedFunction<ResultSet, T, SQLException>
     ): Stream<T> {
-        val connection = dataSource.connection
-        return JdbcDatabase.Companion.toUnsafeStream<T>(
+        var connection = dataSource.connection
+
+        try {
```

Comment:
Sure, I have changed the connection to a val. The code won't be changed to connection.use as we discussed since the connection needs to be open when the result set is returned